### ISSUE TITLE
[TensorFlow 1.14] : Use tf.data.TFRecordDataset to obtain hdfs data, and the network can reach a maximum of 600Mb/s. However, the network is dual 10G.

### ISSUE THREAD

**Issue by Mr-Nineteen** [tensorflow/tensorflow#42315]
1. description：
    **Machine environment**: 4U40C128G.
    When num_parallel_calls is set to 4, 10, 32, and the network  can reach 600Mb/s. The parameter(num_parallel_calls) does not take effect. 
    At the same time, the network card is dual 10G, using hadoop fs -copyToLocal, the network can reach 2g/s.
    Using tf.data.TFRecordDataset to load hdfs data, the network did not reach the bottleneck, and cores did not reach the bottleneck(Use less than 5 cores).
2. code
def input_fn(input_files, num_epochs=1, shuffle=True, batch_size=1000, buffer_size=100000):
    dataset = tf.data.TFRecordDataset(input_files)
    dataset = dataset.batch(batch_size)
    dataset = dataset.prefetch(buffer_size=batch_size * 10)
    dataset = dataset.repeat(num_epochs)
    dataset = dataset.map(decode, num_parallel_calls=32)
    iterator = dataset.make_one_shot_iterator()
    example,lables = iterator.get_next()
    return example,lables

pls why?
Is it a performance problem with TensorFlow loading data, or is it the wrong way to use it?

**Comment by Saduf2019** [comment]
@Mr-Nineteen 
I ran the code and did not face any issue, please find [gist here](https://colab.research.google.com/gist/Saduf2019/cf34c7cc45b3bde7a768d516f425728c/untitled370.ipynb).
Can you please provide with a colab gist the error faced for us to analyse. (share error logs if any)

**Comment by Mr-Nineteen** [comment]
@Saduf2019 

       Machine environment: **4U40C128G**, the network card is **dual 10G**.
 
       Using hadoop fs -copyToLocal, the network can reach 2g/s. Using tf.data.TFRecordDataset to load hdfs data, the network did not reach the bottleneck, and cores did not reach the bottleneck(Use less than 5 cores).

       Give you the code, the machine environment and the HDFS data environment are also different.

       Part of the reference code:

def input_fn(input_files, num_epochs=1, shuffle=True, batch_size=1000, buffer_size=100000):
  dataset = tf.data.TFRecordDataset(input_files)
  dataset = dataset.batch(batch_size)
  dataset = dataset.prefetch(buffer_size=batch_size * 10)
  dataset = dataset.repeat(num_epochs)
  dataset = dataset.map(decode, num_parallel_calls=32)
  iterator = dataset.make_one_shot_iterator()
  example,lables = iterator.get_next()
  return example,lables

def feed_dict_gen(self, features):        
        feed_dict = {
            self.labels_ctr: features['label_ctr'],
            self.rt_user_cate3id_len: features['user_cat3id_hist_len'],
            Omit.......
        }
        return feed_dict

features, labels_ctr  = inputs.input_fn(fnames, num_epochs=1, shuffle=False, batch_size=FLAGS.batch_size)
cur_feed_dict_ = model.feed_dict_gen(features)

 with tf.Session() as sess:
        sess.run(init_op)
        step = 0
                
        try:
            coord = tf.train.Coordinator()
            threads = tf.train.start_queue_runners(coord=coord)
            
            while not coord.should_stop():
                print("load data start: %s step %d"%(datetime.datetime.now(), step))
                cur_feed_dict = sess.run(cur_feed_dict_)
                print("load data end  : %s step %d, batch_size = %d"%(datetime.datetime.now(), step, FLAGS.batch_size))             
                step = step + 1
        except tf.errors.OutOfRangeError as e:
            print("%s except out of range." % (datetime.datetime.now()))
            print(e)
        finally:
            coord.request_stop()
            print('all threads are asked to stop!')

**Comment by Saduf2019** [comment]
@Mr-Nineteen 
Please try using 2.1 as this version is not supported, and let us know if you face any error with tf 2.x
